DashAI/back/plugins/utils.py

```diff
@@ -147,12 +116,91 @@ def install_plugin_from_pypi(pypi_plugin_name: str) -> None:
     RuntimeError
         If pip install command fails
     """
+    pip_action = "install" if install else "uninstall"
+    args = ["pip", pip_action, pypi_plugin_name]
+    args if install else args.append("-y")
+
     res = subprocess.run(
-        ["pip", "install", pypi_plugin_name],
+        args,
         stderr=subprocess.PIPE,
         text=True,
     )
     if res.returncode != 0:
         errors = [line for line in res.stderr.split("\n") if "ERROR" in line]
         error_string = "\n".join(errors)
         raise RuntimeError(error_string)
+
+
+def install_plugin(plugin_name: str) -> List[type]:
+    """
+    Install and register new plugins in component registry
+
+    Parameters
+    ----------
+    plugin_name : str
+        A string with the name of the plugin in pypi to install
+
+    component_registry : ComponentRegistry
+        The current app component registry
+
+    """
+    pre_installed_plugins: List[type] = get_available_plugins()
+    install_plugin_from_pypi(plugin_name)
+    installed_plugins = set(get_available_plugins()) - set(pre_installed_plugins)
+    return installed_plugins
+
+
+def register_plugin_components(installed_plugins, component_registry):
+    for plugin in installed_plugins:
+        try:
+            component_registry.register_component(plugin)
+        except Exception as e:
+            logging.exception(f"Plugin {plugin} could not be registered. Error: {e}")
+
+
+def uninstall_plugin(
+    plugin_name: str,
+) -> List[type]:
+    """
+    Uninstall an existing plugin and delete it from component registry
+
+    Parameters
+    ----------
+    plugin_name : str
+        A string with the name of the plugin in pypi to install
+
+    component_registry : ComponentRegistry
+        The current app component registry
+
+    """
+    available_plugins: List[type] = get_available_plugins()
+    install_plugin_from_pypi(plugin_name, False)
+    uninstalled_components: List[type] = set(available_plugins) - set(
```

Comment:
Here you are retrieving the set of the installed components!
```suggestion
    installed_components: List[type] = set(available_plugins) - set(
```

Reply:
Are you sure? I think that we are taking the set of installed plugins pre-uninstallation, and then we are removing the available plugins post uninstallation.

`
uninstalled_components: List[type] = set(available_plugins) - set(
        get_available_plugins()
    )
`